### PR TITLE
fix: Moved Databricks removal into explicit removal (before RG) for managed resource removal

### LINKED ISSUE #1581: [AVM Question/Feedback]: Databricks Connector not correctly removed
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Description

The `databricks/workspace` module creates an additional `-managed` resource group that is not removed by the CI. This can cause issues upon re-run if the location selector choses a different location to deploy to:

Example:
image

This means we need to investigate why this is the case, if there is a property we can set (similar to how you can specify on the VM that it should remove its NIC & PIP) or whether we can add something to the removal logic (e.g. adding it to the order in which resources are removed - to before the resource group is removed). 
```[tasklist]
### Tasks
```

### PR DESCRIPTION
**Pull request by AlexanderSehr**
## Description

Moved Databricks removal into explicit removal (before RG) for managed resource removal

Fixes #1581

## Pipeline Reference

| Pipeline |
| -------- |
|   [avm.res.databricks.workspace](https://github.com/AlexanderSehr/bicep-registry-modules/actions/workflows/avm.res.databricks.workspace.yml)       |

## Type of Change

- [x] Update to CI Environment or utlities (Non-module effecting changes)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `version.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `version.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `version.json`.
  - [ ] Update to documentation

### CODE REVIEW

File: avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1 [hunk 1]
```diff
@@ -92,6 +92,7 @@ function Initialize-DeploymentRemoval {
             'Microsoft.Compute/virtualMachines',
             'Microsoft.VirtualMachineImages/imageTemplates', # Must be removed before their MSI
             'Microsoft.ManagedIdentity/userAssignedIdentities',
+            'Microsoft.Databricks/workspaces',
```
Comment (eriqua) [inline avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:95]: Tried and failed hard

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:95]: Like the pipeline failed? Or the RGs remained? My 3 tests so far worked, but each time the Databricks deployment itself had issues

Comment (eriqua) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:95]: The RG remained. Let's maybe see if the behavior changed and this workaround works now. If still not, as discussed, we can work on a custom post removal logic.

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:95]: Re-running one last time in the hopes that it's not getting stuck on the dependency issue again - like for the last 4 times.

Comment (AlexanderSehr) [reply avm/utilities/pipelines/e2eValidation/resourceRemoval/Initialize-DeploymentRemoval.ps1:95]: Can't get around the issue during the deployment. Will close the PR for now

**Pull request closed without merging**